### ISSUE TITLE
I'm Up All Night To Get Lucky - The Four Leaf Coffee Brewer

### ISSUE THREAD

**Issue by New-Writer** [CleverRaven/Cataclysm-DDA#23129]
## **Proposal:**

Add a rare coffee machine variant based on the real life Clover brewing machine [used in Starbucks Reserve cafes.](https://1912pike.com/high-definition-starbucks-clover-brewer/) Known as the Four Leaf, to avoid copyright issues.

The real life machine is really cool to think about in a near future context, could add to C:DDA's setting in a small but fun way, and would make coffee shops a more attractive target to visit in the early to mid-game.

## **Lore:**

**(These first paragraphs are the real life justification for this proposal - What the Clover is, why Starbucks bought it, etc. It's all IRL information and not super exciting unless you're really into coffee. If you want to skip right to the C:DDA part, there's a TL;DR in bold.)**

For the last decade or so, larger brands like Budweiser have been losing market share to their smaller counterparts. Craft beer has taken off like a rocket, "fast casual" restaurants are stealing market share from fast food, and coffee giants like Nestle and Starbucks have faced serious competition from third wave coffee shops. In most cases, the bigger brands just bought their competitors and called it a day. Nestle bought Blue Bottle Coffee, Peet's bought Stumptown and Intelligensia, etc. Usual corporate stuff.

Starbucks, the absolute madman, actually tried to compete in terms of quality. They bought the Coffee Equipment Company in 2008, famous for producing an $11,000 ultra-futuristic machine used in only a few elite American cafes. That machine was the Clover, and it already functioned like something out of Cataclysm.

-Cut brewing time from 4 minutes to 30-45 seconds with no loss in quality.
-Algorithmic temperature control, stopping it from fluctuating even one degree.
-Control over every aspect of the process - e.g. weight of coffee grounds used *within a tenth of a gram.*
-Connection to a private network called CloverNet, where other users could save their brew settings as "templates" and share them freely for other cafes to use. 
-Almost full automation. Just stir the water for a few seconds and clean it off with a squeegee.
-[Makes a cool futuristic noise for no reason at all.](https://www.youtube.com/watch?v=3dek2lKcSCQ&t=1m14s)

The third wave coffee community was shocked that Starbucks wanted anything to do with this thing. Then they were sad, because Starbucks did nothing with it for half a decade - and wasn't letting other cafes buy them. Finally, they were *confused.* New "Starbucks Reserve" locations were opening up in major cities to give the machine a "theater of brewing," CloverNet had been expanded massively to collect data on customer satisfaction, and the machine itself... Was the same. No change at all. 

What were they *thinking?*

Something pretty devious, as it turns out. But first the TL;DR.

TL;DR: $11,000 automatic coffee machine invented, company that makes it is bought by Starbucks. They don't use it for half a decade, then launch a special line of stores to showcase it and equip the machine with rudimentary datamining support.

**(Cataclysm info starts here!)**

In hindsight, it's pretty obvious what was going on. 

Starbucks - Err, I mean Cargo House - built Reserve Cafes in cities full of coffee lovers - Seattle, San Francisco, etc. Those cafes had machines which could take customer satisfaction data and send it in aggregate it back to Corporate HQ. Then HQ could push an update, and everyone's lattes are brewed one degree hotter. Three-tenths of a gram less beans in the espresso, but only on the west coast. Our flagship store needs a better star rating on Shout - What brewing template has the highest satisfaction ranking, on average? 

You get the idea. 

Third wave aficionados couldn't help themselves from visiting - *Cargo House* making premium coffee? *That* ought to be good! - and inevitably posted blogs, tweets, and reviews about precisely what was wrong with their java. A few of them were baristas for rival cafes, and wrote lengthy critiques about what could be done better. Internet Service Providers happily forked over the search histories of their users (Congress had given the okay in 2017) and both Facebook and Google were overjoyed to play ball. It was enough raw data to make your eyes water, all diligently sifted through by predictive algorithms and sorted into a neat set of handy correlations.
 
Coffee was becoming a solved game.

It took a few years, but joyous ridicule turned to uneasy appreciation. Reserve Cafes were getting better fast, customers noted, and nobody could explain why. Not even former baristas could solve the mystery. Yes, Cargo House was steadily automating more of the process - the beans were now ground and portioned out by machine - but they were still the ones pushing the buttons! It wasn't like the machines were deciding for themselves!

Slowly but surely, in a true sign of the apocalypse, Cargo House came to serve the best coffee in America. 

This wasn't the end of indie coffee, although it certainly could have been. There were rumors about why they weren't pushing harder, most of them claiming it was by government mandate. Close all the small shops, they said, and unemployment goes through the roof. The real conspiracy nuts went even further, claiming the third wave shops were integral to Cargo House's success. They needed a control group, the rumors went! This was all some damn experiment! Nobody listened, of course. What was so evil about good tasting coffee?

Around a year before the Cataclysm hit, Cargo House was in talks with a *lot* of shady individuals. Rivtech executives looking to optimize the caffeine intake of their employees, or microdose them without their knowledge. XEDRA front companies asking to sponsor special drinks, containing "proprietary nutritional additives" to increase the health of the drinker. A few administrators working in 4th axis labs wanted machines with direct access to CloverNet - err, "CoffeeNet" - just for bragging rights.  Did Cargo House encourage this cooperation, or was the military industrial complex forcing them to play ball? Your guess is as good as mine. But some of that stuff did end up in CoffeeNet, walled off behind various passwords and identity checks.

## **Gameplay:**

Upon mapgen, there is a small chance a spawned coffee shop will contain a Four Leaf brewer instead of a regular coffee machine. It is slightly harder to use, but is worth the effort.

-**Four Leaf machines can only be fed with roasted coffee beans, not coffee powder.** The machine has to control each part of the process! They are slightly more resource heavy to compensate for their increased quality.

-**It's possible to use them in the dark, because the process is automatic. But you still have to stand there and act like you're doing something,** just like the baristas did.

-**The machine is noisy.** Mostly for show. Making noise in the middle of town wasn't always matter of life or death.

-Without some way to identify yourself to the machine - Cell phone? Laptop? Self Awareness trait? CBM - it can't connect to CoffeeNet and automatically decide what your near-perfect cup would be. Instead, **it gives you a choice to brew one of four unique coffee variants.** The most popular Cargo House blends in New England, statistically speaking.

-Sunrise Espresso, a drink with much higher enjoyment and slightly higher nutrition than normal coffee... With almost no quench. Most people drinking espresso aren't doing it because they're thirsty
-Hand-Crafted Cappuccino, identical to a regular coffee with higher enjoyment. Drinking it removes the coffee and adds "Hand-Crafted Cappuccino Foam" to your inventory, a rapidly perishable food item with no health or quench but moderate enjoyment. 
-Enorme Parisian Roast, an otherwise normal coffee with higher than normal enjoyment and two uses. It's huge!
-Seasonal Spice Delight, a Junk Food item with no health or quench, but high enjoyment. Let's be honest. It's just ice cream and whip in a coffee cup.

Actually connecting to CoffeeNet, if such a thing is possible, could function like a Mr. Stem Cell machine. In exchange for *lots* of coffee beans, coffee syrup, and clean water, you get a turbo-enjoyable cup of coffee that gives a lengthy mood boost, a temporary bonus to action points, or some other unique bonus. Nothing so great that savescumming it is gamebreaking.

One sixth of the time, the coffee is *too* perfect. The machine tapped into the aggregate data for all of North America and created a hot cup of ambrosia! You hallucinate and potentially gain a stimulant addiction. There are things man was not meant to know. Or taste.

**Comment by illi-kun** [comment]
Imo, this looks like a very specific request with a minor game effect. Or as a Starbucks ad.

**Comment by New-Writer** [comment]
You could change the company from a Starbucks expy to a fictional one and it wouldn't make a difference to me! I just figured a nationwide, self updating coffee machine botnet was a pretty far fetched (and silly) concept, and it would need really strong justification to make sense in the lore. The Clover machines were the closest I could come while staying realistic. 

It's definitely not intended as an ad, either. Some big company quietly stealing trade secrets, operating as a de-facto NSA, and either planning to quietly mutate the world or being forced into it by XEDRA is pretty dystopian! I'm really into third wave coffee, and Starbucks (or anyone) crushing all the tiny stores this way would be pretty sad.

It's definitely a minor effect though! The TL;DR is something like "Harder to use, loud coffee machine with fun item descriptions and better effects. Also you can roll the dice to get super coffee or go insane." You could make it a Rivtech prototype or add this functionality into the Eternal Boiler/Atomic Coffee Machine and I'd still be happy.

**Comment by Mecares** [comment]
Sounds like a Mod only idea to me

**Comment by Leland** [comment]
I mean no offense because there's clearly been quite a bit of work put into this, but this proposal reads like an advertisement.

**Comment by New-Writer** [comment]
No offense taken. There’s a thin line between using actual companies to justify fiction and accidental product placement. I don’t like the brand of coffee I’m referencing at all, but it’s not like there’s any way for someone playing the game to know that.

If you think the basic idea of a coffee machine variant is solid, we could throw out the fluff I wrote and assume every high end coffee shop has a machine with that level of precision. Making them a rare spawn, but not associated with any company in particular. If that’s too minor to be worth doing, this request can be closed.

**Comment by kevingranade** [comment]
My only concern is that attaching a conspiracy theory to the backstory  might give someone a reason to take legal interest in the project.  Definitely disassociate it from Starbucks, but other than that, no problem.

I think people might be misconstruing the extent of the suggested content, this amounts to a new location with a unique piece of furniture, and a handful of drink items.  If it's more than that, i.e. snippets or quests, those are things we could use a lot more of.

**Comment by New-Writer** [comment]
If that new location is a generic high end coffee shop with a special machine, there are a ton of potential history snippets and quests that could spawn from it.

-Coffee snob NPC in the refugee center is willing to pay in items if you find one of these stores and get him his favorite drink. He supplies a thermos to keep it warm once you pour it.

-Old Guard representative confides that they’re facing a serious morale problem, but one of their intelligence agents has a solution: go to this coffee shop, load this software onto the machine, and test it to make sure it works. Then bring it back. The software makes it brew a *too* perfect cup every time.

-NPC’s kid is having a birthday. It could seem silly given the situation, but they’ll pay you to pick up some gifts to brighten the kid’s day. They’d pick three items at random from a list of toys, junk food, and books. The super sweet coffee could be one potential item.

-You could play around with funny spawns, like corpses lined up at the counter still waiting for a drink, child zombies with three empty coffee cups in their inventory, a NPC behind the counter who treats you like a customer until you snap him out of it (at which point he acts like a normal NPC), etc.

-Flyers and newspaper snippets hinting at what was going on. “Machine learning experts horrified as independent cafes fight a war of automation.” “Coffee experts claim the fourth wave has arrived.” “Food reviewer in coma after ingesting cup of AI-brewed java.” “Rivtech denies dosing employees with ‘Atomic Coffee,’ calls rumored effects of the drink, ‘Ludicrous, if not scientifically impossible.’” “Checkmate? Robotic coffee machine scores perfect marks at Tokyo coffee festival.”

-If other coffee brewing items are ever added, like a French Press or hand grinder, this location would be a good place to spawn them.

**Comment by New-Writer** [comment]
A potential way to save time adding the building to mapgen: Lots of independent coffee shops roast their own beans, and have a door in the back of their cafe that leads to a roastery. Rather than designing a new building from scratch, you could take the existing coffee shop floorplan and add a small back room.

Here are a few examples of attached roasteries taken from Reddit: https://imgur.com/a/TIYyX

**Comment by Night-Pryanik** [comment]
More than half a year of no discussion, closing.
